docs/DesignDocs/AttestationAPIsWithFlexibleClaims.md

```diff
@@ -0,0 +1,571 @@
+Attestation APIs with Init- and Run-Time Claims Support
+==========================
+
+Motivation
+------------
+
+The Open Enclave SDK currently supports attestation APIs,
+`oe_get_attestation_certificate_with_evidence_v2` and
+`oe_verify_attestation_certificate_with_evidence_v2`, that
+allow developers to obtain a self-signed certificate with
+the enclave evidence embedded and to verify the certificate,
+respectively. The latter API also outputs an array of claims,
+allowing the developers to implement additional verification
+logic. However, these APIs have following limitations:
+
+- Lack of support of init-time custom claims
```

Comment:
Init-time claims can be entered at the enclave creation time `oe_create_enclave` and stays the same for the life time of the enclave. It is supported. No need to add extra parameters for it here.

Reply:
CONFIG_DATA (or CONFIG_ID) is used to verify the integrity of the init-time claims. See example in
https://github.com/openenclave/openenclave/blob/master/docs/DesignDocs/InitTimeConfigurationInterface.md#plugin-api-change-with-explicit-init-time-custom-claim-buffer-support

```
    /// Verifier Plugin
    ///
    oe_result_t oe_verify_evidence(
    const oe_uuid_t* format_id,
    const uint8_t* evidence_buffer,
    size_t evidence_buffer_size,
    const uint8_t* endorsements_buffer,
    size_t endorsements_buffer_size,
    const oe_policy_t* policies,
    size_t policies_size,
    oe_claim_t** claims,
    size_t* claims_length)
    {
        ...
        //extract the init_time_claims_buffer and verify it
        inittime_claims_buffer = _find_init_time_claim_buffer(evidence_buffer,
                                   evidence_buffer_size);
        inittime_claims_buffer_size = _find_init_time_claim_buffer_size(
                                        evidence_buffer, evidence_buffer_size);
        config_id = _find_config_id(evidence_buffer, evidence_buffer_size);
        if (inittime_claim_buffer.integrity_algorithm_id ! = 0 )
            return error;
        if (SHA256(inittime_claim_buffer.buffer, inittime_claims_buffer_size)
              != config_id)
            return error;
        //output init_time claims
        ...
    }

```